[FtheoryTools] Added F-theory models on all toric hypersurfaces

emikelsons opened the pull request:
1. I'm still not sure about the correct discete factors for the gauge groups, F12 I think I figured out, but I'm not sure about F14

2. For models 13,15 16 where the MW group has torsion, do we need to list the generators which correspond to the torsion part?(Also in the paper they only list the Weierstrass coordinates for these generators)

emikelsons committed:
[FtheoryTools] Added F-theory models on all toric hypersurfaces

emikelsons commented:
cc @HereAround @apturner

HereAround commented:
Thank you @emikelsons  for this great first PR!

Suggestion: Please let us add CI tests (see the `test/literature_models.jl` file).

Regarding your above questions, let us hear from @apturner to see how we should best proceed.

HereAround committed:
[FTheoryTools] Change base dimension for 1408_4808 models from 2 to 3

HereAround committed:
[FTheoryTools] Minor changes on some 1408_4808 models so that we can create a new index

HereAround committed:
[FTheoryTools] Create new literature model index

HereAround committed:
[FTheoryTools] Move Weierstrass forms, currently indistinguishable for our index, to separate folder

HereAround committed:
[FtheoryTools] Add simple test for F1 hypersurface model in 1408_4808

HereAround commented:
@emikelsons and @apturner , as mentioned in slack, I ahve taken the liberty to take the next step in this PR. Hence, I began by adding CI-tests. Here is a summary of my actions and observations:

* Rebased to current master.
* Weierstrass forms of the newly added 1408_4808 models cannot be distinguished by our current index from their hypersurface counterparts. The information in the model json-files that is used to create the model index is identical. Hence, the index needs to be expanded and/or the model information. For now, I moved the Weierstrass counterparts to a separate folder, so they are not considered for the index. (Otherwise, we get an error "could not uniquely identify the model", which effectively means that we cannot even create a single new model.) This is of course only a temporary fix, so that I could demonstrate how the CI tests can be added.
* A couple of comments (related to the questions above by @emikelsons) prevented the creation of a new model index - the json files could then not be read-in to create the index. I have adjusted this, while making sure that we do not forget about these issues.
* Created a new model index.
* I noticed that the F1-model could not be created, because it lacks the information about the divisor classes D1 and D2. Those are the equivalent of D and \tilde(D) on page 17 of https://arxiv.org/pdf/1408.4808.pdf. I have (or do hope to have) added this information correctly. If I have not made a mistake, it now says that D1 (a.k.a. D in the notation of https://arxiv.org/pdf/1408.4808.pdf) is the divisor (class) of the model section s7, and likewise D2 (a.k.a. \tilde(D) in notation of https://arxiv.org/pdf/1408.4808.pdf) is the divisor (class) of the model section s9. With that change, the model could be created.
* For time reasons, I have not looked through all the other models. But they might also suffer from not having the D1, D2 information. This needs to be looked into.
* I noticed that the model description for the F1-model was very generic, and so I adjusted that too. This should also be checked for the other models.
* Finally, I added a first preliminary test for the F1-model. This test could be expanded. Also, similar tests should be added for all the remaining 1408_4808 models.

But first, let us see if the CI-tests complete with the changes I made.

HereAround committed:
[FTheoryTools] Extend test for F1 hypersurface model in 1408_4808

HereAround commented:
I have just rebased to the latest master and extended the F1-test by also construction the F1 model over a concrete base. Hopefully, this is a useful foundation on which we can discuss the next steps in a few minutes during our zoom meeting @emikelsons and @apturner .

apturner commented on experimental/FTheoryTools/src/LiteratureModels/Models/model1408_4808-12.json:
```diff
@@ -0,0 +1,70 @@
+        "type": "hypersurface",
+        "description": "The toric variety associated to F_12",
+        "gauge_algebra": ["su2","su2","u1","u1"],
+        "global_gauge_quotients": [["Z4", [1, 2, 3, 4]]]
```
Based on Fig. 21 and Table 22, it seems this should actually be `"global_gauge_quotients": [["Z2", [1, 3]], ["Z2", [2, 4]]]`, and the same is true for the WSF version

HereAround commented on experimental/FTheoryTools/test/literature_models.jl:
```diff
@@ -367,3 +367,37 @@ h = literature_model(arxiv_id = "1507.05954", equation = "3.4")
+  @test dim(base_space(foah1)) == 3
+  @test dim(ambient_space(foah1)) == 5
+  @test is_base_space_fully_specified(foah1) == true
+  @test model_description(foah1) == "F-theory hypersurface model with fiber ambient space F_1"
```
```suggestion
  @test dim(base_space(foah1_B3)) == 3
  @test dim(ambient_space(foah1_B3)) == 5
  @test is_base_space_fully_specified(foah1_B3) == true
  @test model_description(foah1_B3) == "F-theory hypersurface model with fiber ambient space F_1"
```

(My mistake earlier, but did not want to push again, as this would likely cause more merge conflicts for Mikelis.)

HereAround replied:
This mistake of mine causes some (if not all) of the test failures.

apturner commented on experimental/FTheoryTools/src/LiteratureModels/Models/model1408_4808-15.json:
```diff
@@ -0,0 +1,70 @@
+        "type": "hypersurface",
+        "description": "The toric variety associated to F_15",
+        "gauge_algebra": ["su2", "su2", "su2", "su2", "u1"],
+        "global_gauge_quotients": [["Z2", [1, 2, 3, 4]]]
```
After further inspection, I believe this should be `[["Z2", [1, 2, 3, 4]], ["Z2", [1, 4, 5]]]`

apturner commented on experimental/FTheoryTools/src/LiteratureModels/Models/model1408_4808-9.json:
```diff
@@ -0,0 +1,70 @@
+        "type": "hypersurface",
+        "description": "The toric variety associated to F_9",
+        "gauge_algebra": ["su2", "u1", "u1"],
+        "global_gauge_quotients": [["Z2", [1, 2]]]
```
In the order presented in the paper, this should be `[["Z2", [1, 3]]]`

emikelsons committed:
Update experimental/FTheoryTools/test/literature_models.jl

Co-authored-by: Martin Bies <HereAround@users.noreply.github.com>

apturner commented on experimental/FTheoryTools/src/LiteratureModels/Models/model1408_4808-14.json:
```diff
@@ -0,0 +1,69 @@
+        "type": "hypersurface",
+        "description": "The toric variety associated to F_14",
+        "gauge_algebra": ["su3","su2","su2","u1"],
+        "global_gauge_quotients": [["Correct?"]]
```
As we discussed, I would put `[["Z3", [1, 4]], ["Z2", [2, 3, 4]]]` here

apturner commented on experimental/FTheoryTools/src/LiteratureModels/Models/model1408_4808-11.json:
```diff
@@ -0,0 +1,69 @@
+        "type": "hypersurface",
+        "description": "The toric variety associated to F_11",
+        "gauge_algebra": ["su3","su2","u1"],
+        "global_gauge_quotients": [["Z6", [1, 2, 3]]]
```
As discussed, this might be more clear as `[["Z3", [1, 3]], ["Z2", [2, 3]]]`

emikelsons committed:
Added changes that were discussed

1)Added related models in addition to associated models
2)Added correct gauge quotients
3)Added a more detailed model description for all other models
4)Added divisor classes D1, D2 for all other models
5)Added generic coordinates for torsion generators

apturner reviewed (approved):
This looks good @emikelsons, thanks for all the work! The coordinate changes for the base gradings look good on models where you had to do that.

Really, we should make a new field to store the coordinates of the torsional generating sections, something like "torsion_sections", as we want to distinguish them from the U(1) generating sections. That could either be done now or in a subsequent PR.

HereAround commented:
(I have just approved the run of the tests. This is likely necessary because this is @emikelsons first PR to OSCAR. Please ping me in once you want to run the tests again.)

emikelsons committed:
Added CI tests for other models

emikelsons committed:
Added D1,D2 corrections and torsion sections

As discussed I banished F4 to the WSF folder for now. I also spotted that I had forgot to add the WSF coords for the torsion sections of F16(In the paper they only list 1, but since the order is 3, 2*P=-P)

HereAround commented:
Just rebased onto the latest master. Let us see if the test now complete successfully.

HereAround committed:
[FTheoryTools] Update model index

HereAround commented:
Since the F4-model was removed, we also seem to need to update the index, which I have done just now.

HereAround committed:
[FTheoryTools] Disable F4 tests

HereAround commented:
Since the F4-model was moved, we can of course not execute its tests. That is why (or so I believe) the tests failed. I have just disabled the F4 tests.
